src/ui/MLFacet/MLFacet.ts

```diff
@@ -351,6 +364,7 @@ export class MLFacet extends Component {
     Assert.exists(data.queryBuilder);
     const queryBuilder = data.queryBuilder;
     this.mLFacetQueryController.putFacetIntoQueryBuilder(queryBuilder);
+    this.mLFacetQueryController.setFreezeCurrentValuesFlag(false);
```

Comment:
I think it would be better if the "reset" mechanism could somehow be handled internally in the `mLFacetQueryController` class, using the "duringQuery" event I mentioned.

If we were able to have a structure where any piece of code (custom or other components) can put the flag to true, and not have to bother with resetting it manually afterwards, it would be better I think.